contracts/LifCrowdsale.sol

```diff
@@ -191,18 +210,41 @@ contract LifCrowdsale is Ownable, Pausable {
 
     uint256 tokens = weiSent.mul(privatePresaleRate);
 
-    require(totalPresaleWei.add(weiSent) <= maxPresaleWei);
-
     totalPresaleWei.add(weiSent);
 
     token.mint(beneficiary, tokens);
   }
 
   // send ether to the fund collection wallet
-  function forwardFunds() onlyOwner {
-    foundationWallet.transfer(this.balance);
-    // TODO
-    // marketMaker.transfer(marketMakerPercentage * this.balance);
+  function forwardFunds() internal {
+
+    // calculate the max amount of wei for the foundation
+    uint256 foundationBalanceCapWei= foundationBalanceCapUSD.mul(weiPerUSD);
+
+    // if the minimiun cap for the market maker is not reached transfer all funds to foundation
+    // else if the min cap for the market maker is reached, create it and send the remaining funds
+    if (this.balance < foundationBalanceCapWei) {
+
+      foundationWallet.transfer(this.balance);
+
+    } else {
+
+      uint256 mmFundBalance = this.balance.sub(foundationBalanceCapWei);
+
+      // check how much preiods we have to use on the market maker
+      uint8 marketMakerPeriods = 24;
+      if (mmFundBalance > marketMaker24PeriodsCapUSD.mul(weiPerUSD))
+        marketMakerPeriods = 48;
+
+      foundationWallet.transfer(foundationBalanceCapWei);
+      LifMarketMaker marketMaker = new LifMarketMaker(
+        address(token), block.number.add(10), 20, marketMakerPeriods, foundationWallet, 105000
```

Comment:
we should add a variable for the delay between this finalize and the market maker startBlock. I see you are using `10` but we talked about using 2 weeks, remember? At least please add a TODO here

Reply:
:+1: